File: Exercise/LeetCode/Array/P27Pro.java
package Array;

public class P27Pro {
    public int removeElement(int[] nums, int val) {

        int leftIndex = 0;
        int rightIndex = nums.length - 1;
        while (leftIndex <= rightIndex) {
            while (leftIndex <= rightIndex && nums[leftIndex] != val) {
                leftIndex++;
            }

            while (leftIndex <= rightIndex && nums[rightIndex] == val) {
                rightIndex--;
            }

            if (leftIndex <= rightIndex) {
                nums[leftIndex++] = nums[rightIndex--];
            }
        }

        return leftIndex;
    }

    public static void main(String[] args) {

    }
}

File: Exercise/LeetCode/Array/P27Pro.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;
class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        int leftIndex = 0;
        int rightIndex = nums.size() - 1;
        while (leftIndex <= rightIndex)
        {
            while (leftIndex <= rightIndex && nums[leftIndex] != val)
            {
                leftIndex++;
            }

            while (leftIndex <= rightIndex && nums[rightIndex] == val)
            {
                rightIndex--;
            }

            if (leftIndex <= rightIndex)
            {
                nums[leftIndex++] = nums[rightIndex--];
            }
        }

        return leftIndex;
    }
};
int main(void)
{

    return 0;
}